src/r_draw.c

```diff
@@ -526,21 +526,185 @@ static void R_DrawFuzzColumn_block(void)
+    4 * FUZZPAL,0,0,6 * FUZZPAL,6 * FUZZPAL,6 * FUZZPAL,6 * FUZZPAL,0,
+    4 * FUZZPAL,0,6 * FUZZPAL,6 * FUZZPAL,0,0,6 * FUZZPAL,
+    6 * FUZZPAL,0,0,0,0,6 * FUZZPAL,6 * FUZZPAL,
+    6 * FUZZPAL,6 * FUZZPAL,0,6 * FUZZPAL,6 * FUZZPAL,0,6 * FUZZPAL,
+};
+
+static void DrawFuzzColumnRefraction(void)
+{
+    boolean cutoff = false;
 
-void R_SetFuzzColumnMode(void)
+    if (dc_x % fuzzblocksize)
+    {
+        return;
+    }
+
+    if (!dc_yl)
+    {
+        dc_yl = 1;
+    }
+
+    if (dc_yh == viewheight - 1)
+    {
+        dc_yh = viewheight - 2;
+        cutoff = true;
+    }
+
+    int count = dc_yh - dc_yl;
+
+    if (count < 0)
+    {
+        return;
+    }
+
+#ifdef RANGECHECK
+    if ((unsigned)dc_x >= video.width || dc_yl < 0 || dc_yh >= video.height)
+    {
+        I_Error("R_DrawFuzzColumn: %i to %i at %i", dc_yl, dc_yh, dc_x);
+    }
+#endif
+
+    ++count;
+
+    byte *dest = ylookup[dc_yl] + columnofs[dc_x];
+
+    int lines = fuzzblocksize - (dc_yl % fuzzblocksize);
+
+    int dark = FUZZDARK;
+    int offset = 0;
+
+    do
+    {
+        count -= lines;
+
+        // if (count < 0)
+        // {
+        //    lines += count;
+        //    count = 0;
+        // }
+        const int mask = count >> (8 * sizeof(mask) - 1);
+        lines += count & mask;
+        count &= ~mask;
+
+        const byte fuzz = fullcolormap[dark + dest[linesize * offset]];
+
+        do
+        {
+            memset(dest, fuzz, fuzzblocksize);
+            dest += linesize;
+        } while (--lines);
+
+        ++fuzzpos;
+
+        // Clamp table lookup index.
+        fuzzpos &= (fuzzpos - FUZZTABLE) >> (8 * sizeof(fuzzpos) - 1); // killough 1/99
+
+        dark = fuzzdark[fuzzpos];
+        offset = fuzzoffset[fuzzpos];
+
+        lines = fuzzblocksize;
+    } while (count);
+
+    if (cutoff)
+    {
+        const byte fuzz =
+            fullcolormap[dark + dest[linesize * (offset - FUZZOFF) / 2]];
+        memset(dest, fuzz, fuzzblocksize);
+    }
+}
+
+static void DrawFuzzColumnShadow(void)
 {
-    if (fuzzcolumn_mode && current_video_height > SCREENHEIGHT)
+    boolean cutoff = false;
+
+    // Adjust borders. Low...
+    if (!dc_yl)
     {
-        fuzzblocksize = FixedToInt(video.yscale);
-        R_DrawFuzzColumn = R_DrawFuzzColumn_block;
+        dc_yl = 1;
     }
-    else
+
+    // .. and high.
+    if (dc_yh == viewheight - 1)
+    {
+        dc_yh = viewheight - 2;
+        cutoff = true;
+    }
+
+    int count = dc_yh - dc_yl;
+
+    // Zero length.
+    if (count < 0)
+    {
+        return;
+    }
+
+#ifdef RANGECHECK
+    if ((unsigned)dc_x >= video.width || dc_yl < 0 || dc_yh >= video.height)
     {
-        R_DrawFuzzColumn = R_DrawFuzzColumn_orig;
+        I_Error("R_DrawFuzzColumn: %i to %i at %i", dc_yl, dc_yh, dc_x);
+    }
+#endif
+
+    byte *dest = ylookup[dc_yl] + columnofs[dc_x];
+
+    count++; // killough 1/99: minor tuning
+
+    do
+    {
+        *dest = fullcolormap[12 * 256 + *dest];
```

Comment:
Let's make this a little brighter and it'll be perfect: `8 * 256 ...`

Reply:
Done in d0d203b